### PR TITLE
P4 2185 filter reports

### PR DESCRIPTION
**Pull request by jcavell**
@matt-crowson-moj smaller PR this time, to filter out unwanted moves, journeys and events based on their status / event type.

Also catch parsing errors and log a warning, so it carries on processing reporting lines.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporter.kt [hunk 1]
```diff
@@ -24,6 +24,7 @@ class ReportingImporter {
                 moveFiles = movesContent,
                 journeyFiles = journeysContent,
                 eventFiles = eventsContent,
+//                eventFiles = listOf(),
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporter.kt:27]: Remove this if redundant.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporterTest.kt [hunk 2]
```diff
@@ -15,7 +15,7 @@ import java.time.LocalDate
 @ExtendWith(SpringExtension::class)
 @Import(TestConfig::class)
 @ActiveProfiles("test")
-//@ActiveProfiles("localstack")
+//@ActiveProfiles("dev")
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporterTest.kt:18]: Is the commented out stuff supposed to be here?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporterTest.kt [hunk 3]
```diff
@@ -33,8 +33,8 @@ internal class ReportingImporterTest {
     fun `Get files for date should ignore missing days`() {
         val content = importer.import(LocalDate.of(2020, 9, 3))
 
-        // There should be 4 unique moves
-        Assertions.assertEquals(4, content.size)
+        // There should be 3 moves over the two files that are rejected or cancelled
+        Assertions.assertEquals(3, content.size)
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporterTest.kt:37]: Is there anything else we can assert on here other than just the size?

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporterTest.kt (hunk 2 above)
Comment (jcavell) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporterTest.kt:18]: Yeah...still switching between different profiles when testing. Maybe I could have conditional logic, I'll look at that

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporterTest.kt (hunk 3 above)
Comment (jcavell) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporterTest.kt:37]: Yep

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporter.kt (hunk 1 above)
Comment (jcavell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/ReportingImporter.kt:27]: Having to cater for staging not having event reports in the correct (v2) format

**Pull request merged**